Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/FunctionBlock.java

```diff
@@ -0,0 +1,259 @@
+    @FXML private Pane outputSpace;
+
+    /** The space containing all the arguments of the function. */
+    private ArgumentSpace argumentSpace;
+
+    /** The space containing the output anchor of this FunctionBLock. */
+    @FXML private Pane outputTypesSpace;
+
+    /**
+     * Method that creates a newInstance of this class along with it's visual
+     * representation
+     *
+     * @param name
+     *            The name of the function.
+     * @param type
+     *            The function's type (usually a FuncT).
+     * @param pane
+     *            The parent pane in which this FunctionBlock exists.
+     */
+    public FunctionBlock(String name, Type type, CustomUIPane pane) {
+        super(pane);
+        this.name = new SimpleStringProperty(name);
+
+        this.loadFXML("FunctionBlock");
+
+        // Collect argument types
+        ArrayList<String> args = new ArrayList<>();
+        Type t = type;
+        while (t instanceof FuncT) {
+            FuncT ft = (FuncT) t;
+            args.add(ft.getArgs()[0].toHaskellType());
+            t = ft.getArgs()[1];
+        }
+        
+        List<Type> inputSignatures = new ArrayList<>();
+        Type functionSignature = getFunctionSignature();
+        for (int i = 0; i < args.size(); i++) {
+            if (functionSignature instanceof FuncT) {
+                Type inputSignature = BackendUtils.dive((FuncT) functionSignature, i + 1);
+                inputSignatures.add(inputSignature);
+            } else {
+                throw new FunctionDefinitionException();
+            }
+        }
+        
+        argumentSpace = new ArgumentSpace(this, inputSignatures);
+        argumentSpace.setKnotIndex(getAllInputs().size());
+        
+        ((Pane) this.lookup("#nestSpace")).getChildren().add(argumentSpace);
+        argumentSpace.knotIndexProperty().addListener(e -> invalidateKnotIndex());
+        
+        // Create an anchor for the result
+        output = new OutputAnchor(this);
+        output.layoutXProperty().bind(outputSpace.widthProperty().divide(2));
+        outputSpace.getChildren().add(output);
+        
+        // Make sure the prefWidth is correctly updated.
+        Pane functionInfo = (Pane) this.lookup("#functionInfo");
+        this.prefWidthProperty().bind(functionInfo.widthProperty().add(argumentSpace.prefWidthProperty()));
+        
+        
+        this.setMinSize(Region.USE_PREF_SIZE, Region.USE_PREF_SIZE);
+        this.setMaxSize(Region.USE_PREF_SIZE, Region.USE_PREF_SIZE);
+        functionInfo.setMinWidth(Region.USE_PREF_SIZE);
+        functionInfo.setMaxWidth(Region.USE_PREF_SIZE);
+        invalidateConnectionState();
+    }
+    
+    /** Returns the name property of this FunctionBlock. */
+    public final String getName() {
+        return name.get();
+    }
+
+    /** Sets the name property of this FunctionBlock. */
+    public void setName(String name) {
+        this.name.set(name);
```

Comment:
The FunctionBlock's name label is binded to the nameProperty. To my understanding we then should implement all 3 methods.